### PR TITLE
Add initial support for RoCE packets

### PR DESCRIPTION
**Pull request by haggaie**
This PR adds BTH headers and CNP sub-headers, with a few basic tests. We can add other headers (RDMA, acks, or datagrams) later on.

I've added a way to read hexdumps from a string to make it easier to write the test. If there's a better way to do that I'd be happy to learn.

I also wasn't sure whether this belongs in layers or contrib.

### CODE REVIEW

**Comment by haggaie** [comment]
Hi, I'm not sure the build failure [here](https://travis-ci.com/github/secdev/scapy/jobs/298199907) is related to this patch. The failure is in an "obdscanner" test.

File: scapy/config.py [hunk 1]
```diff
@@ -634,7 +634,7 @@ class Conf(ConfClass):
                    'inet6', 'ipsec', 'ir', 'isakmp', 'l2', 'l2tp',
                    'llmnr', 'lltd', 'mgcp', 'mobileip', 'netbios',
                    'netflow', 'ntp', 'ppi', 'ppp', 'pptp', 'radius', 'rip',
-                   'rtp', 'sctp', 'sixlowpan', 'skinny', 'smb', 'snmp',
+                   'roce', 'rtp', 'sctp', 'sixlowpan', 'skinny', 'smb', 'snmp',
```
Comment (guedou) [inline scapy/config.py]: Your layer should do to `contrib/`. The goal is to keep to minimize the number of layer loaded by default.

File: scapy/layers/roce.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            payload = raw(pshdr[BTH].payload)
+            # add ICRC placeholder just to get the right IP.totlen and
+            # UDP.length
+            icrc_placeholder = b'\xff\xff\xff\xff'
+            pshdr[UDP].payload = Raw(bth + payload + icrc_placeholder)
+            icrc = crc32(raw(pshdr)[:-4]) & 0xffffffff
+            return self.pack_icrc(icrc)
+        else:
+            # TODO support IPv6
+            warning("No IP underlayer to compute checksum. Leaving null.")
```
Comment (guedou) [inline scapy/layers/roce.py]: You should display a warning saying that the `underlayer` protocol is not supported.

File: scapy/utils.py [hunk 3]
```diff
@@ -1534,16 +1534,20 @@ def _write_packet(self, packet, sec=None, usec=None, caplen=None,
 
 
 @conf.commands.register
-def import_hexcap():
+def import_hexcap(s=None):
```
Comment (guedou) [inline scapy/utils.py]: Could you use a meaningful variable name such as `input_string` and add a parameter documentation as done here https://github.com/secdev/scapy/blob/master/scapy/utils.py#L251?

**Comment by haggaie** [comment]
Thanks for your comments. I'll update the PR accordingly.

**Comment by gpotter2** [comment]
Could you rebase against master? thanks

**Comment by haggaie** [comment]
> Could you rebase against master? thanks

No problem.

**Comment by haggaie** [comment]
I checked the errors in the Travis CI run. It seems like the only error is in a test called `obdscanner` [here](https://travis-ci.com/github/secdev/scapy/jobs/314199654#L6169). I don't think it is related to this pull request.

File: scapy/config.py (hunk 1 above)
Comment (haggaie) [reply scapy/config.py]: Done.

File: scapy/utils.py (hunk 3 above)
Comment (haggaie) [reply scapy/utils.py]: Done. I hope I did it correctly.

File: scapy/layers/roce.py (hunk 2 above)
Comment (haggaie) [reply scapy/layers/roce.py]: Done. I tried adding the underlayer name as well to make the warning clear.

**Comment by guedou** [comment]
Sorry for the delay in merging your PR!

**Pull request merged**